src/app/authentication/auth.service.ts

```diff
@@ -1,46 +1,81 @@
 import { Injectable } from '@angular/core';
+import * as JWT from 'jwt-decode';
 import { AppComponent } from '../app.component';
 
 declare let Msal: any;
+
 @Injectable()
 export class AuthService {
+
     private app: any;
 
     constructor() {
-        this.app = new Msal.UserAgentApplication((window as any).ClientId, '', this.authCallback, {});
-    }
-
-    public authCallback = (errorDesc, token, error) => {
-        if (token) {
-            localStorage.setItem('status', 'authenticated');
-        } else if (errorDesc || error) {
-            localStorage.setItem('status', 'anonymous');
-        }
+        this.app = new Msal.UserAgentApplication((window as any).ClientId, '',
+            {});
     }
```

Comment:
We can make this code more readable and easy to understand by splitting the retrieving of the `ClientId` and initialization of the `UserAgentApplication` like below.

```
const { ClientId } = (window as any)
this.app = new Msal.UserAgentApplication(ClientId. '', {});

```